### PR TITLE
[NL DateTime] Initial support of Dutch DateTimeModel in .NET 

### PR DESCRIPTION
**Pull request by iMicknl**
- First version of DateTimeModel, containing Date + DateTime + Holidays

**current status:**
DateExtractor:  ❌(61 of 113 still unsupported)
DateParser: ❌(61 of 113 still unsupported)
DateTimeExtractor: ❌(73 of 111 still unsupported)
HolidayParser: ✅ (1 of 12 unsupported)
HolidayExtractor: ✅ (9 supported)

### CODE REVIEW

**Comment by iMicknl** [comment]
@tellarin done! However it looks like many .cs files just had a small change, due to the new method of generating resources. I will have a look if I can reproduce this in master and create a new PR with only those files, otherwise I will revert it.

File: .NET/Microsoft.Recognizers.Text.DateTime/DateTimeRecognizer.cs [hunk 1]
```diff
@@ -97,12 +98,21 @@ protected override void InitializeConfiguration()
                         new GermanMergedParserConfiguration(new BaseOptionsConfiguration(options))),
                     new BaseMergedDateTimeExtractor(new GermanMergedExtractorConfiguration(options))));
 
+            // TODO to be uncommented when all tests for Dutch are green.
+            RegisterModel<DateTimeModel>(
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/DateTimeRecognizer.cs]: This should be kept commented out before we can merge the PR.

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchSetParserConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            }
+            else if (trimmedText.Equals("wekelijks"))
+            {
+                timex = "P1W";
+            }
+            else if (trimmedText.Equals("tweewekelijks"))
+            {
+                timex = "P2W";
+            }
+            else if (trimmedText.Equals("maandelijks"))
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchSetParserConfiguration.cs]: Can you move these hard coded strings to the YAML resources? Some languages have this done partially already.

File: .NET/Samples/SimpleConsole/Program.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ namespace SimpleConsole
     class Program
     {
         // Use English for the Recognizers culture
-        private const string DefaultCulture = Culture.English;
+        private const string DefaultCulture = Culture.Dutch;
```
Comment (tellarin) [inline .NET/Samples/SimpleConsole/Program.cs]: Be careful not to change the culture in the Sample.

File: Specs/DateTime/Dutch/HolidayParser.json [hunk 4]
```diff
@@ -271,33 +215,46 @@
         "Start": 15,
         "Length": 26
       }
-    ],
-    "NotSupported": "dotNet"
+    ]
   },
   {
-    "Input": "Ik ga terug op thanksgiving day 2010",
```
Comment (tellarin) [inline Specs/DateTime/Dutch/HolidayParser.json]: Was this a completely invalid sentence? We should keep as many as possible for coverage if useful.

File: Specs/DateTime/Dutch/TimeZoneParser.json [hunk 5]
```diff
@@ -224,7 +214,6 @@
     "Context": {
```
Comment (tellarin) [inline Specs/DateTime/Dutch/TimeZoneParser.json]: Is "pacific daylight saving time" how you'd typically reference such timezone in a Dutch sentence?

**Comment by tellarin** [comment]
@iMicknl We can merge this once the main changes are in and conflicts resolved. Thanks.

**Comment by tellarin** [comment]
> DateTimeExtractor: ❌(111 of 73 still unsupported)

I don't think this is right. ;)

**Comment by iMicknl** [comment]
@tellarin thanks for the review. I will try to fix it end of week and clean up this branch.

File: .NET/Samples/SimpleConsole/Program.cs (hunk 3 above)
Comment (iMicknl) [reply .NET/Samples/SimpleConsole/Program.cs]: #resolved

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchSetParserConfiguration.cs (hunk 2 above)
Comment (iMicknl) [reply .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchSetParserConfiguration.cs]: Will do in a new PR, where I also fix the [EnglishSetParserConfiguration.cs](https://github.com/Microsoft/Recognizers-Text/blob/master/.NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishSetParserConfiguration.cs). Most of my files are just a 1:1 copy of the English version at the moment.

File: .NET/Microsoft.Recognizers.Text.DateTime/DateTimeRecognizer.cs (hunk 1 above)
Comment (iMicknl) [reply .NET/Microsoft.Recognizers.Text.DateTime/DateTimeRecognizer.cs]: #resolved

**Comment by iMicknl** [comment]
@tellarin any idea why the private CI is failing? I have no insights unfortunately. 

I processed some of your comments, I would prefer to do other comments in different PR's. Just break it into smaller chunks.

**Comment by tellarin** [comment]
> @tellarin any idea why the private CI is failing? I have no insights unfortunately.

The build was actually failing in AppVeyor too, but was not breaking due to an error in the build script.
You can see the log now there.

The other items can be fixed in follow-up PRs. No problem.

Thanks!

**Comment by iMicknl** [comment]
@tellarin fixed the tests. Will work on the other comments in follow-up PR's. Would be great if this one could me merged in before I have to rebase / refactor again.

**Pull request merged**